packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -872,6 +1012,128 @@ All @auth directives used on field definitions are performed when the field is r
         }
     }
 
+    // OnCreate Subscription
+    private protectOnCreateSubcription(ctx: TransformerContext, rules: AuthRule[],
+        parent: ObjectTypeDefinitionNode, nameOverride?: string) {
+        const fieldName = nameOverride ? nameOverride : graphqlName(ON_CREATE_FIELD + toUpper(parent.name.value));
+        this.protectSubscription(ctx, rules, parent, fieldName)
+    }
+
+    // OnUpdate Subscription
+    private protectOnUpdateSubcription(ctx: TransformerContext, rules: AuthRule[],
+        parent: ObjectTypeDefinitionNode, nameOverride?: string) {
+        const fieldName = nameOverride ? nameOverride : graphqlName(ON_UPDATE_FIELD + toUpper(parent.name.value));
+        this.protectSubscription(ctx, rules, parent, fieldName)
+    }
+
+    // OnDelete Subscription
+    private protectOnDeleteSubcription(ctx: TransformerContext, rules: AuthRule[], parent: ObjectTypeDefinitionNode, nameOverride?: string) {
+        const fieldName = nameOverride ? nameOverride : graphqlName(ON_DELETE_FIELD + toUpper(parent.name.value));
+        this.protectSubscription(ctx, rules, parent, fieldName)
+    }
+
+    // generate subscrption based on operation
+    private protectSubscription(ctx: TransformerContext, rules: AuthRule[],
+        parent: ObjectTypeDefinitionNode, fieldName: string) {
+        const resolverResourceId = ResolverResourceIDs.ResolverResourceID("Subscription", fieldName);
+        const resolver = this.resources.generateSubscriptionResolver(fieldName);
+
+        // creates the none data source
+        const noneDS = ctx.getResource(ResourceConstants.RESOURCES.NoneDataSource)
```

Comment:
Do we want to create a data source if its not being used? Should you set this only if there is `rules.length > 0`

Reply:
Creating a none data source if the subscription resolvers are being created.